tests/Fixer/Import/NoDuplicatedImportsFixerTest.php

```diff
@@ -0,0 +1,170 @@
+
+/**
+ * @internal
+ *
+ * @covers \PhpCsFixer\Fixer\Import\NoDuplicatedImportsFixer
+ */
+final class NoDuplicatedImportsFixerTest extends AbstractFixerTestCase
+{
+    /**
+     * @dataProvider provideFixCases
+     */
+    public function testFix(string $expected, ?string $input = null): void
+    {
+        $this->linter = new Linter();
+        $this->doTest($expected, $input);
+    }
+
+    public function provideFixCases(): iterable
+    {
+        yield 'same import but in different namespace' => [
+            '<?php
+namespace {
+    use Bar;
+
+    echo Bar::foo;
+}
+
+namespace A {
+    use Bar;
+
+    echo Bar::foo;
+}
+            ',
+        ];
+
+        yield 'multiple duplicates' => [
+            '<?php use A;      ',
+            '<?php use A;use A;use A;use A;use A;use A;use A;',
+        ];
+
+        yield 'no leading - no leading' => [
+            '<?php
+use Throwable1;
+ ',
+            '<?php
+use Throwable1;
+use Throwable1;',
+        ];
+
+        yield 'leading - leading' => [
+            '<?php
+use \Throwable2;
+ ',
+            '<?php
+use \Throwable2;
+use \Throwable2;',
+        ];
+
+        yield 'leading - no leading' => [
+            '<?php
+use \Throwable3;
+ ',
+            '<?php
+use \Throwable3;
+use Throwable3;',
+        ];
+
+        yield 'no leading - leading' => [
+            '<?php
+use Throwable4;
+ ',
+            '<?php
+use Throwable4;
+use \Throwable4;',
+        ];
+
+        yield 'do not remove close tag' => [
+            '<?php
+use Throwable5;
+  ?> Foo
+',
+            '<?php
+use Throwable5;
+use Throwable5 ?> Foo
+',
+        ];
+
+        yield 'comments are preserved' => [
+            '<?php use A;/* A *//* B *//* C *//* D */ // foo',
+            '<?php use A;/* A */use/* B */A/* C */;/* D */ // foo',
+        ];
+
+        yield 'function' => [
+            '<?php use function A;   ',
+            '<?php use function A; use function A;',
+        ];
+
+        yield 'const' => [
+            '<?php use const A;   ',
+            '<?php use const A; use const A;',
+        ];
+
+        yield 'scattered' => [
+            '<?php declare(strict_types=1);namespace B; use const A; echo 123; $a = time(); if ($a % 2 == 1) { echo "foo"; };   ?>X<?php echo 4567;',
+            '<?php declare(strict_types=1);namespace B; use const A; echo 123; $a = time(); if ($a % 2 == 1) { echo "foo"; }; use const A?>X<?php echo 4567;',
+        ];
+
+        yield 'mixed import types |' => [
+            '<?php
+             use A;
+             use const A;',
+        ];
+
+        yield 'mixed import types ||' => [
+            '<?php
+             use const A;
+             use function A;',
+        ];
+
+        yield 'mixed import types |||' => [
+            '<?php
+             use A;
+             use const A;
+             use function A;',
+        ];
+
+        yield 'whatever' => [
+            '<?php
+namespace FooBar;
+use Vendor\Project\Duplicated\Foo;
+use Vendor\Foo;
```

Comment:
This `Foo` is still a duplicate, will cause a [fatal error](https://3v4l.org/skmEP):

> Fatal error: Cannot use Vendor\Foo as Foo because the name is already in use

Reply:
The fixer cannot know which of them to drop.
duplicated FQCN imports were removed. can't do more here

Reply:
Message from fatal error says which one cannot be used because the other exists.

> duplicated FQCN imports were removed

There is no such thing as "FQCN imports" - imported is class, not FQCN, FQCN is a path/place where to import that class from.

With keeping duplicates the name of the fixer would be misleading, it will remove duplicated FQCN only, not classes.

Reply:
> Message from fatal error says which one cannot be used because the other exists.

but you know how the file potentially ends up with having such "duplicated Foo" imports, right? most scenario is git conflict. keeping the "first in appearance order" as "proper" is not correct. compiler doesn't know which is dev has to solve the conflict. deleting "all but first" is doing harm, let's not go that way


> There is no such thing as "FQCN imports" 

seriously? like... seriously? I'm more than happy to hear how you prefer to name the difference of those imports here, but to be honest i don't think it matters how you name it.


if we have
```
use \a\b\c;
use \a\b\c;
```
we _for sure_ know that one of them can be removed, and as they are exactly the same (FQCN is the same), we remove any of those lines.


if we have
```
use \a\x;
use \b\x;
```

we have no clue which `x` is correct to use. let's not touch it



file after applying this fixer will not be perfect, but will be better than it was

Reply:
> you know how the file potentially ends up with having such "duplicated Foo" imports, right?

Yes, I still remember that I've explained the reason for having this fixer in the issue 😉 

> deleting "all but first" is doing harm, let's not go that way

I disagree on that. Many risky fixers (like `declare_strict_types` or `date_time_immutable`) will break a codebase if it is not ready for such fix.

> > There is no such thing as "FQCN imports"
> 
> seriously? like... seriously?

Yes, like YES. It's actually very similar to real world. If you have 2 bananas in your fridge and one comes from Ecuador and the other from Costa Rica are the different fruit? Because their "FCQN" is Banana from Ecuador and Banana from Costa Rica? No, both are just banana in your fridge.

> if we have
> 
> ```
> use \a\x;
> use \b\x;
> ```
> 
> we have no clue which `x` is correct to use. let's not touch it

But I - as a developer and I believe I'm not alone on that - want fix errors "Fatal error: Cannot use blah blah because the name is already in use".

Maybe at least an option like `remove_duplicated_imports_even_when_from_different_namespace`?